src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySinkConfig.java

```diff
@@ -149,4 +193,101 @@ void validateCmekKey(FailureCollector failureCollector, Map<String, String> argu
     }
     validateCmekKeyLocation(cmekKeyName, null, location, failureCollector);
   }
+
+  public String getDataset() {
+    return dataset;
+  }
+
+  public String getDatasetProject() {
+    if (connection == null) {
```

Comment:
I think connection will always have value, we made it backward compatible if connection was not provided , we get it from parent properties

Reply:
also connection is null should not be equivalent to auto-detect dataset project

Reply:
In configure time, the connection will be null if user chooses to use connection and that should be the only case connection will be null.
I think if connection is null, we can just return null instead of the default project id.

Reply:
done